todo_app/data/utils.py

```diff
@@ -0,0 +1,15 @@
+from typing import Any, Callable
+
+import requests
+
+from todo_app.data.exceptions import ResponseError
+
+
+def catch_response_failure(func: Callable[[Any], requests.Response]):
+    def wrap(*args, **kwargs):
```

Comment:
Worth noting that it's good practice to use the `@wraps` decorator when creating your own - [more info on this SO thread for why & how exactly](https://stackoverflow.com/questions/308999/what-does-functools-wraps-do)